src/components/crossDevice/CrossDeviceLink/index.js

```diff
@@ -246,8 +246,11 @@ class CrossDeviceLinkUI extends Component {
     const mobileUrl = cross_device_url || hosted_sdk_url
     // This lets us test the cross device flow locally and on Surge.
     // We use the same location to test the same bundle as the desktop flow.
+
+    // TODO: review this change
+
     return process.env.MOBILE_URL === '/'
-      ? `${window.location.origin}?link_id=${this.linkId}`
+      ? `${window.location.origin}${window.location.pathname}?link_id=${this.linkId}`
```

Comment:
The mobile url was always generated without the pathname. Running locally under http://localhost/somedir/ ended up in a link generated http://localhost/?link_id skipping the path.

/cc @ethan605

Reply:
In which cases did you need to use the pathname locally? Anyways, the change looks good to me

Reply:
For local testing without the need to rebuild the SDK. Before, we always assumed, that the cross origin url was served under http://host:port/.